examples/05_stable_diffusion/src/modeling/resnet.py

```diff
@@ -47,21 +48,29 @@ def __init__(
 
         conv = None
         if use_conv_transpose:
-            conv = nn.ConvTranspose2dBias(channels, self.out_channels, 4, 2, 1)
+            conv = nn.ConvTranspose2dBias(
+                channels, self.out_channels, 4, 2, 1, dtype=dtype
+            )
         elif use_conv:
-            conv = nn.Conv2dBias(self.channels, self.out_channels, 3, 1, 1)
+            conv = nn.Conv2dBias(self.channels, self.out_channels, 3, 1, 1, dtype=dtype)
 
         # TODO(Suraj, Patrick) - clean up after weight dicts are correctly renamed
         if name == "conv":
             self.conv = conv
         else:
             self.Conv2d_0 = conv
 
-    def forward(self, x):
+    def forward(self, x, upsample_size=None):
         if self.use_conv_transpose:
             return self.conv(x)
-
-        x = nn.Upsampling2d(scale_factor=2.0, mode="nearest")(x)
+        out = None
+        if upsample_size is not None:
+            out = ops.size()(x)
+            out[1] = upsample_size[1]
+            out[2] = upsample_size[2]
+            out = [x._attrs["int_var"] for x in out]
+            out = Tensor(out)
+        x = nn.Upsampling2d(scale_factor=2.0, mode="nearest")(x, out)
```

Comment:
`test_correctness.py` isn't set up for XL, so can you clarify whether you made relevant changes, or were you testing v1? I will run the tests myself and report back.
Regarding #875, community has discovered that with XL and 8px increment, specifically resolution that is only multiple of 8 and not multiple of 16 or 32, images are not correct, the issue is documented [here](https://github.com/FizzleDorf/AIT/issues/30). v1 appears to be unaffected. If `test_correctness.py` is indeed failing this is likely the issue.
As for whether this PR should be merged independently, I can reverse the #875 specific changes in this PR or this PR can wait until I've resolved the issue with XL and 8px, I do not mind, whichever makes it easier to review on your end.

Reply:
I'm running `test_correctness.py` as it is in this PR (i.e. same as on the base commit). The test indeed doesn't use XL, but, since you are modifying code shared between XL and v1, it's important to make sure existing tests don't break. On my side it breaks on the assert discussed above

Reply:
That's right, it was just the assert that was breaking it - removing it should be enough. If you would like to reopen this PR, I'll import it and rerun the tests